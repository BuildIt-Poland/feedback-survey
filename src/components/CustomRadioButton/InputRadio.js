import styled from 'styled-components';

const InputRadio = styled.input`
  height: 0;
  width: 0;
  display: none;
`;

export default InputRadio;
