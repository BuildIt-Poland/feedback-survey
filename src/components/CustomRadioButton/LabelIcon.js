import styled from 'styled-components';

const LabelIcon = styled.label`
  :hover {
    cursor: pointer;
  }

  img {
    width: 8rem;
  }

  span {
    display: none;
  }
`;

export default LabelIcon;
