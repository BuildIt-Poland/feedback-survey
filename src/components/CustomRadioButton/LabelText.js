import styled from 'styled-components';

import media from '../../styles/media';
import { colorWhite, colorBlue } from '../../styles/designTokens';

const LabelText = styled.label`
  width: 15rem;
  height: 4.8rem;
  line-height: 4.8rem;
  text-transform: lowercase;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  margin: 0;
  border-radius: 1rem;

  ${media.phone`
    width: 25%;
    max-width: 15rem;
  `};

  input:not(:checked) + & {
    color: ${colorBlue};
    border: ${props => (props.withIcon ? 'none' : `0.2rem ${colorBlue} solid`)};
  }

  input:not(:checked) + &:hover,
  input:checked + & {
    color: ${colorWhite};
    background-color: ${props => (props.withIcon ? 'none' : `${colorBlue}`)};
    border: ${props => (props.withIcon ? 'none' : `0.2rem ${colorBlue} solid`)};
  }
`;

export default LabelText;
