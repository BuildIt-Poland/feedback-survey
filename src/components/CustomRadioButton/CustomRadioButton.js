import React from 'react';
import { bool, string } from 'prop-types';

import { radioOption } from '../../types';
import InputRadio from './InputRadio';
import LabelIcon from './LabelIcon';
import LabelText from './LabelText';

const renderLabel = (label, id, withIcon) => {
  if (withIcon) {
    return <LabelIcon htmlFor={id} label={label} />;
  }

  return <LabelText htmlFor={id}>{label}</LabelText>;
};

const CustomRadioButton = ({ radio, name, choosenValue, onChange, withIcon = false }) => {
  const { value, label, id } = radio;
  const isChecked = value === choosenValue;
  return (
    <>
      <InputRadio
        withIcon={withIcon}
        type="radio"
        name={name}
        value={value}
        onChange={onChange}
        id={id}
        checked={isChecked}
      />
      {renderLabel(label, id, withIcon)}
    </>
  );
};

CustomRadioButton.propTypes = {
  radio: radioOption.isRequired,
  name: string.isRequired,
  withIcon: bool
};

export default CustomRadioButton;
