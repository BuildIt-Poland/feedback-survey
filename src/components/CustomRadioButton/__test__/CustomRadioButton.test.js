import React from 'react';
import { create } from 'react-test-renderer';

import CustomRadioButton from '../CustomRadioButton';

describe('COMPONENT - CustomRadioButton', () => {
  it('renders correctly', () => {
    const component = create(<CustomRadioButton />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
