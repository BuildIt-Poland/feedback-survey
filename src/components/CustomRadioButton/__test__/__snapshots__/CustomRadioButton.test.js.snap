// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - CustomRadioButton renders correctly 1`] = `
Array [
  <input
    className="sc-bdVaJa hoFWaF"
    type="radio"
  />,
  <label
    className="sc-htpNat dkbVkt"
  />,
]
`;
